cpp_guidelines/.clang-format

```diff
@@ -0,0 +1,107 @@
+---
+Language: Cpp
+Standard: c++17
+DisableFormat:   false
+AccessModifierOffset: -4
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: true
+AlignConsecutiveBitFields: true
+AlignConsecutiveDeclarations: false
+AlignConsecutiveMacros: true
+AlignEscapedNewlines: Right
+AlignOperands: true
+AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: false
+AllowAllConstructorInitializersOnNextLine: false
+AllowAllParametersOfDeclarationOnNextLine: false
+AllowShortBlocksOnASingleLine: Never
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortEnumsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLambdasOnASingleLine: All
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterReturnType: None
+AlwaysBreakBeforeMultilineStrings: false
+AlwaysBreakTemplateDeclarations: Yes
+BinPackArguments: false
+BinPackParameters: false
+BitFieldColonSpacing: Both
+BreakBeforeBinaryOperators: All
+BreakBeforeBraces: Allman
+BreakBeforeTernaryOperators: true
+BreakConstructorInitializers: BeforeComma
+BreakInheritanceList: BeforeComma
+BreakStringLiterals: true
+ColumnLimit:     79
+CommentPragmas:  '^ COMMENT pragma:'
+CompactNamespaces: false
+ConstructorInitializerAllOnOneLineOrOnePerLine: true
+ConstructorInitializerIndentWidth: 4
+ContinuationIndentWidth: 4
+Cpp11BracedListStyle: true
+DeriveLineEnding: false
+DerivePointerAlignment: false
+ExperimentalAutoDetectBinPacking: false
+FixNamespaceComments: true
+IncludeBlocks:   Preserve
+IncludeCategories:
+  - Regex:           '^"(llvm|llvm-c|clang|clang-c)/'
+    Priority:        2
+    SortPriority:    0
+  - Regex:           '^(<|"(gtest|gmock|isl|json)/)'
+    Priority:        3
+    SortPriority:    0
+  - Regex:           '.*'
+    Priority:        1
+    SortPriority:    0
+IncludeIsMainRegex: '(Test)?$'
+IncludeIsMainSourceRegex: ''
+IndentCaseBlocks: true
+IndentCaseLabels: false
+IndentExternBlock: AfterExternBlock
+IndentGotoLabels: true
+IndentPPDirectives: AfterHash
+IndentWidth: 4
+IndentWrappedFunctionNames: false
+KeepEmptyLinesAtTheStartOfBlocks: false
+MacroBlockBegin: ''
+MacroBlockEnd:   ''
+MaxEmptyLinesToKeep: 2
+NamespaceIndentation: All
+OperandAlignmentStyle: Align
+PenaltyBreakAssignment: 2
+PenaltyBreakBeforeFirstCallParameter: 19
+PenaltyBreakComment: 300
+PenaltyBreakFirstLessLess: 120
+PenaltyBreakString: 1000
+PenaltyBreakTemplateDeclaration: 10
+PenaltyExcessCharacter: 1000000
+PenaltyReturnTypeOnItsOwnLine: 1000
+PointerAlignment: Left
+ReflowComments: true
+SortIncludes: true
+SortUsingDeclarations: true
+SpaceAfterCStyleCast: true
+SpaceAfterLogicalNot: false
+SpaceAfterTemplateKeyword: true
+SpaceBeforeAssignmentOperators: true
+SpaceBeforeCpp11BracedList: true
+SpaceBeforeCtorInitializerColon: true
+SpaceBeforeInheritanceColon: true
+SpaceBeforeParens: ControlStatements
```

Comment:
**VOTE**
```
current | Never |
--------|-------------|
 0      |   1         |
```

IMO space after `if`, `for` looking wired. 

# Example:
current setting:
```C++
{
    auto x = !line;
    if (line.m_pos0[0] == line.m_pos1[0])
        return;

    for (int j = T_begin; j < T_end; ++j)
    {
        for (int i = T_begin; i < T_end - 1; ++i)
        {
            const float_X W = this->DS(line, i, 0) * tmp;
            accumulated_J += W;

        }
    }
}
```

Suggestion:
```C++
{
    auto x = !line;
    if(line.m_pos0[0] == line.m_pos1[0])
        return;

    for(int j = T_begin; j < T_end; ++j)
    {
        for(int i = T_begin; i < T_end - 1; ++i)
        {
            const float_X W = this->DS(line, i, 0) * tmp;
            accumulated_J += W;
        }
    }
}
```

Reply:
current | Never |
--------|-------------|
 1      |   1         |

I prefer the space before control statements, but it is not a strong preference. Can I just give half a vote for current?

Reply:
current | Never |
--------|-------------|
 0      |   2        |

Reply:
current | Never |
--------|-------------|
 1      |   1         |

Reply:
current | Never |
--------|-----------|
 2      |   3

Reply:
current | Never
-- | --
2 | 4

Reply:
current | Never |
--------|-------------|
 3     |   4         |

Reply:
current | false       |
--------|-------------|
 4     |   4         |

Reply:
current | Never
-- | --
5 | 4

Reply:
Reminder: Voting will be closed on 2020-09-26 at 11:32 HZDR time.

Reply:
current | Never |
--------|-----------|
 5     |   5

Reply:
Voting closed but we have a tie. Does one of the voters want to change his opinion?

Reply:
I am willing to change my vote for one of the east consters to change theirs ;)

Kidding. You can reduce my vote to 0.5 if you want, that would change the result to `Never`.

Reply:
Let me help:
current | Never |
--------|-----------|
 5     |   6

Reply:
Voting closed, the winner is `Never`.